lib/xav/decoder.ex

```diff
@@ -5,65 +5,78 @@ defmodule Xav.Decoder do
 
   @typedoc """
   Supported codecs.
+
+  To get the list of available decoders see `Xav.list_decoders/0`.
   """
-  @type codec() :: :opus | :vp8 | :h264 | :h265
+  @type codec() :: atom()
 
   @type t() :: reference()
 
-  @typedoc """
-  Opts that can be passed to `new/2`.
-  """
-  @type opts :: [
-          out_format: Xav.Frame.format(),
-          out_sample_rate: integer(),
-          out_channels: integer(),
-          out_width: Xav.Frame.width(),
-          out_height: Xav.Frame.height()
-        ]
+  @decoder_options_schema [
+    out_format: [
+      type: :atom,
+      doc: """
+      Output format of the samples.
+
+      If case of video, it's the pixel format. In case of audio, it's the sample format.
+
+      To get the list of supported pixel formats use `Xav.pixel_formats/0`,
+      and for sample formats `Xav.sample_formats/0`.
+      """
+    ],
+    out_sample_rate: [
+      type: :pos_integer,
+      doc: """
+      Audio sample rate.
+
+      If not specified, the sample rate of the input stream will be used.
+      """
+    ],
+    out_channels: [
+      type: :pos_integer,
+      doc: """
+      Number of audio channels.
+
+      If not specified, the number of channels of the input stream will be used.
+
+      Audio samples are always in the packed form -
+      samples from different channels are interleaved in the same, single binary:
+
+      ```
+      <<c10, c20, c30, c11, c21, c31, c12, c22, c32>>
+      ```
+      """
+    ],
+    out_width: [
+      type: :pos_integer,
+      doc: "Scale the output video frame to the provided width."
+    ],
+    out_height: [
+      type: :pos_integer,
+      doc: "Scale the output video frame to the provided height."
+    ]
+  ]
 
   @doc """
   Creates a new decoder.
 
-  `opts` can be used to specify desired output parameters.
-
-  E.g. if you want to change audio samples format just pass:
-
-  ```elixir
-  [out_format: :f32]
-  ```
+  `codec` is any decoders supported by FFmpeg. Note that only audio/video decoders
+  are supported for now.
 
-  or video samples format:
-
-  ```elixir
-  [out_format: :rgb24]
-  ```
-
-  Audio samples are always in the packed form -
-  samples from different channels are interleaved in the same, single binary:
-
-  ```
-  <<c10, c20, c30, c11, c21, c31, c12, c22, c32>>
-  ```
-
-  An alternative would be to return a list of binaries, where
```

Comment:
I would keep this part. It may help understand how audio can be stored in memory